src/amo/components/AddonDetail.js

```diff
@@ -45,6 +49,50 @@ export class AddonDetailBase extends React.Component {
     this.props.setCurrentStatus();
   }
 
+  onTouchStart = (event) => {
+    // We preventDefault so that the image long-press menu doesn't appear.
+    // This unfortunately prevents scrolling :(
+    event.preventDefault();
+    this.props.previewTheme(event.currentTarget);
+  }
+
+  onTouchEnd = (event) => {
+    this.props.resetPreviewTheme(event.currentTarget);
+  }
+
+  headerImage() {
+    const { addon, getBrowserThemeData, i18n } = this.props;
+    const { previewURL, type } = addon;
+    const iconUrl = isAllowedOrigin(addon.icon_url) ? addon.icon_url :
+      fallbackIcon;
+
+    if (type === THEME_TYPE) {
+      const label = i18n.gettext('Press to preview');
+      return (
+        <div
+          className="AddonDetail-theme-header"
+          id="AddonDetail-theme-header"
+          data-browsertheme={getBrowserThemeData()}
+          onTouchStart={this.onTouchStart}
+          onTouchEnd={this.onTouchEnd}
+        >
+          <label className="AddonDetail-theme-header-label" htmlFor="AddonDetail-theme-header">
```

Comment:
It's surprisingly only 79 characters wide, but I believe we have a lint rule to prevent 100+.